src/test/java/com/github/fishio/TestPlayingField.java

```diff
@@ -0,0 +1,339 @@
+
+	private PlayingField field;
+	private Canvas canvas;
+	
+	/**
+	 * Initialises the PlayingField field.
+	 */
+	@Before
+	public void setUp() {
+		this.canvas = Mockito.mock(Canvas.class);
+		field = new SinglePlayerPlayingField(60, canvas);
+	}
+	
+	/**
+	 * Tests the getFPS method.
+	 */
+	@Test
+	public void testGetFPS() {
+		assertEquals(60, field.getFPS());
+	}
+	
+	/**
+	 * Tests the setFPS method.
+	 */
+	@Test
+	public void testSetFPS() {
+		field.setFPS(80);
+		
+		assertEquals(80, field.getFPS());
+	}
+	
+	/**
+	 * Tests the getWidth method.
+	 */
+	@Test
+	public void testGetWidth() {
+		assertEquals(1280, field.getWidth(), 0.0D);
+	}
+	
+	/**
+	 * Tests the getHeight method.
+	 */
+	@Test
+	public void testGetHeight() {
+		assertEquals(670.0, field.getHeight(), 0.0D);
+	}
+	
+	/**
+	 * Tests the getEntities method.
+	 */
+	@Test
+	public void testGetEntities() {
+		assertEquals(1, field.getEntities().size());
+		assertTrue(field.getEntities().get(0) instanceof PlayerFish);
+	}
+	
+	/**
+	 * Tests the getRenderer method.
+	 */
+	@Test
+	public void testGetRenderer() {
+		assertEquals(canvas, field.getRenderer().getCanvas());
+		assertEquals(60, field.getRenderer().getFps());
+	}
+	
+	/**
+	 * Tests the getGameThread method.
+	 */
+	@Test
+	public void testGetGameThread() {
+		assertEquals(GameState.STOPPED, field.getGameThread().getState());
+	}
+	
+	/**
+	 * Tests the startGame method.
+	 */
+	@Test
+	public void testStartGame() {
+		field.startGame();
+		
+		assertTrue(field.getRenderer().isRendering());
+		
+		GameState state = field.getGameThread().getState();
+		assertTrue(state == GameState.RUNNING || state == GameState.STARTING);
+	}
+	
+	/**
+	 * Tests the startGameAndWait method.
+	 */
+	@Test
+	public void testStartGameAndWait() {
+		try {
+			field.startGameAndWait();
+		} catch (InterruptedException e) {
+			fail();
+		}
+		
+		assertTrue(field.getRenderer().isRendering());
+		assertEquals(GameState.RUNNING, field.getGameThread().getState());
+	}
+	
+	/**
+	 * Tests the stopGame method.
+	 */
+	@Test
+	public void testStopGame() {
+		try {
+			field.startGameAndWait();
+		} catch (InterruptedException e) {
+			fail();
+		}
+		field.stopGame();
+		
+		assertFalse(field.getRenderer().isRendering());
+		GameState state = field.getGameThread().getState();
+		assertTrue(state == GameState.STOPPED || state == GameState.STOPPING);
+	}
+	
+	/**
+	 * Tests the stopGameAndWait method.
+	 */
+	@Test
+	public void testStopGameAndWait() {
+		try {
+			field.startGameAndWait();
+			field.stopGameAndWait();
+		} catch (InterruptedException e) {
+			fail();
+		}
+		
+		assertFalse(field.getRenderer().isRendering());
+		assertEquals(GameState.STOPPED, field.getGameThread().getState());
+	}
+	
+	/**
+	 * Tests the add method.
+	 */
+	@Test
+	public void testAdd() {
+		Entity e = Mockito.mock(Entity.class);
+		
+		field.add(e);
+		
+		assertSame(e, field.getEntities().get(1));
+		assertSame(e, field.getDrawables().getFirst());
```

Comment:
Actually, when doing getDrawables() it is important that the added Entity is the first on that list, because of the order of rendering. However it is true that I should use contains for the getEntities(), I will fix that. Good point.

<logic